include/irr/asset/CTexturePacker.h

```diff
@@ -216,53 +206,79 @@ class ICPUTexturePacker : protected ITexturePacker
     STexOffset pack(const ICPUImage* _img, const ICPUImage::SSubresourceRange& _subres)
     {
         //assert(m_megaimg->getCreationParameters().format==_img->getCreationParameters().format);
+        const STexOffset pgtOffset = alloc(_img, _subres);
 
-        const uint32_t texelSz = getTexelOrBlockBytesize(m_physAddrTex->getCreationParameters().format);
-        auto offsets = alloc(_img, _subres);
-        for (const auto& reg : _img->getRegions())
-        {
-            if (reg.imageSubresource.mipLevel<_subres.baseMipLevel || reg.imageSubresource.mipLevel>=(_subres.baseMipLevel+offsets->size()))
-                continue;
-
-            const uint32_t j = reg.imageSubresource.mipLevel-_subres.baseMipLevel;//"relative LoD"
-
-            const core::vector3du32_SIMD texOffset = pageCoords((*offsets)[j], m_pgSzxy);
-            const uint64_t bufOffset = static_cast<uint64_t>(m_physAddrTex->getCreationParameters().extent.width*m_physAddrTex->getCreationParameters().extent.height*texOffset.z + texOffset.y*m_physAddrTex->getCreationParameters().extent.width + texOffset.x) * texelSz;
+        auto extent = _img->getCreationParameters().extent;
+        uint32_t levelCount = 1u;
+        for (uint32_t i = 0u; (extent.width>>(_subres.baseMipLevel+i) >= m_pgSzxy) || (extent.height>>(_subres.baseMipLevel+i) >= m_pgSzxy); ++i)
```

Comment:
why use a loop to determine `levelCount` ?

Reply:
fixed with your fun equation